internal/ast/ast.go

```diff
@@ -88,15 +88,16 @@ type (
 
 	// An SourceQuery node represents a query entry from the source code.
 	SourceQuery struct {
-		Name        string        // name of the query
-		Doc         *CommentGroup // associated documentation; or nil
-		Start       gotok.Pos     // position of the start token, like 'SELECT' or 'UPDATE'
-		SourceSQL   string        // the complete sql query as it appeared in the source file
-		PreparedSQL string        // the sql query with args replaced by $1, $2, etc.
-		ParamNames  []string      // the name of each param in the PreparedSQL, the nth entry is the $n+1 param
-		ResultKind  ResultKind    // the result output type
-		Pragmas     Pragmas       // optional query options
-		Semi        gotok.Pos     // position of the closing semicolon
+		Name               string            // name of the query
+		Doc                *CommentGroup     // associated documentation; or nil
+		Start              gotok.Pos         // position of the start token, like 'SELECT' or 'UPDATE'
+		SourceSQL          string            // the complete sql query as it appeared in the source file
+		PreparedSQL        string            // the sql query with args replaced by $1, $2, etc.
+		ParamNames         []string          // the name of each param in the PreparedSQL, the nth entry is the $n+1 param
+		ParamTypeOverrides map[string]string // map of Go type to override the Pg type of a param
```

Comment:
nit: I think this can be `ParamGoTypes` with a doc of something like:

> map of user-specified Go types to use for the arg.

Reply:
Sounds good, and for the return value I  would then name it accordingly:
```go
ResultGoTypes map[string]string // map of user-specified Go type for the result columns.
```